smbus: Remove syscalls with callbacks

ceolin opened the pull request:
Remove syscalls that allows user threads to set callbacks that will be invoked by the kernel.

Userspace is not trusted we can't allow a user thread set callbacks that will be invoked by the kernel and run with supervisor privileges.

Commit:
smbus: Remove syscalls with callbacks

Remove syscalls that allows user threads to set callbacks that
will be invoked by the kernel.

Userspace is not trusted we can't allow a user thread set callbacks
that will be invoked by the kernel and run with supervisor privileges.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

finikorg reviewed (changes requested):
Some tests are failing, probably need to fix tests for smbus

Commit:
tests: smbus: Fix user mode usage

APIs that set callback don't have syscalls and their
tests have to run in supervisor mode.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

finikorg commented on include/zephyr/drivers/smbus.h:
```diff
@@ -660,11 +657,8 @@ static inline int z_impl_smbus_smbalert_remove_cb(const struct device *dev,
-static inline int z_impl_smbus_host_notify_set_cb(const struct device *dev,
-						  struct smbus_callback *cb)
+static inline int smbus_host_notify_set_cb(const struct device *dev,
+				       struct smbus_callback *cb)
```
Please fix indentation.

ceolin replied:
these places mixing tab/space ...

finikorg replied:
This is standard coding style (at least used to be), the same like in Linux.

finikorg commented on include/zephyr/drivers/smbus.h:
```diff
@@ -606,11 +606,8 @@ static inline int z_impl_smbus_get_config(const struct device *dev,
-static inline int z_impl_smbus_smbalert_set_cb(const struct device *dev,
-					       struct smbus_callback *cb)
+static inline int smbus_smbalert_set_cb(const struct device *dev,
+				    struct smbus_callback *cb)
```
indentation

finikorg reviewed (changes requested):
Maybe you can also delete remove callbacks syscalls

ceolin commented:
> Maybe you can also delete remove callbacks syscalls

maybe ... but it is not a problem and there are other similar places in Zephyr that allows removing callbacks from usermode. But you are the owner, if you think it is better not having it I can remove them.

finikorg reviewed (approved):
LGTM, let's leave "remove callbacks from user space" if this is standard Zephyr practice.

ceolin commented:
@nashif can you please take a look on it.
